store/displayPreferences.ts

```diff
@@ -1,116 +1,80 @@
-import Vue from 'vue';
 import { DisplayPreferencesDto } from '@jellyfin/client-axios';
-import { MutationTree, ActionTree, GetterTree } from 'vuex';
+import { MutationTree, ActionTree } from 'vuex';
+import { toString as toStr } from 'lodash';
+import { boolean } from 'boolean';
 import nuxtConfig from '~/nuxt.config';
 
-const stringToBoolean = (value: string): boolean => value === 'True';
-const booleanToString = (value: boolean): string => (value ? 'True' : 'False');
-
-export interface DisplayPreferencesState extends DisplayPreferencesDto {
-  CustomPrefs: {
-    [key: string]: string;
-  };
+/**
+ * Casted typings for the CustomPrefs property of DisplayPreferencesDto
+ */
+interface CustomPreferences {
+  darkMode: boolean;
+  locale: string;
+}
+/**
+ * DisplayPreferencesDto but with our custom typings for CustomPrefs property of DisplayPreferencesDto
+ */
+export interface ClientPreferences
+  extends Omit<DisplayPreferencesDto, 'CustomPrefs'> {
+  CustomPrefs?: CustomPreferences;
+}
+export interface DisplayPreferencesState extends CustomPreferences {
+  syncing: boolean;
 }
-
-const defaultCustomPrefs: {
-  [key: string]: string;
-} = {
-  darkMode: booleanToString(
-    nuxtConfig.vuetify?.theme?.default === 'dark' || false
-  ),
-  locale: 'auto'
-};
 
 const defaultState = (): DisplayPreferencesState => ({
-  CustomPrefs: defaultCustomPrefs
+  darkMode:
+    nuxtConfig.vuetify?.theme?.dark !== undefined
+      ? nuxtConfig.vuetify?.theme?.dark
+      : true,
+  locale: 'auto',
+  syncing: false
 });
 
 /**
- * Methods to apply for each CustomPrefs property.
- * Those callbacks are usually done on global variables such as $vuetify or $i18n,
- * and those may not be present early in the application lifecycle.
+ * Convert custom preferences returned from the server in the correct type
+ *
+ * @param {DisplayPreferencesDto} data - Response from the server
+ * @returns {ClientPreferences} - The response with the correct datatypes.
  */
-const updateMethods: { [key: string]: (value: string) => void } = {
-  darkMode: (value: string) => {
-    if (window.$nuxt) window.$nuxt.$vuetify.theme.dark = stringToBoolean(value);
-  },
-
-  locale: (value: string) => {
-    if (window.$nuxt) {
-      if (value !== 'auto') window.$nuxt.$i18n.setLocale(value);
-      else
-        window.$nuxt.$i18n.setLocale(
-          window.$nuxt.$i18n.getBrowserLocale() ||
-            window.$nuxt.$i18n.defaultLocale ||
-            'en'
-        );
-    }
+function castResponse(data: DisplayPreferencesDto): ClientPreferences {
+  const result = data as ClientPreferences;
```

Comment:
This isn't a copy, result is the same object as data

Reply:
Also, `result.CustomPrefs?.darkMode` is still a string after the cast. Gosh JS and types haha

Reply:
@ThibaultNocchi How are you checking that? No issues here.

Reply:
So, as L41 is just a reference copy, I don't think we need to do it

Reply:
![Sin título](https://user-images.githubusercontent.com/10274099/107784354-a0a3db80-6d4b-11eb-9e59-25969da397df.png)

I think you're confusing the use of ``as``. When using ``as``, in "real Javascript" no object casting is happening. In fact, that line is stripped from the compiled code.

That's an abstraction that TypeScript adds you so the object, from that point onwards, is considered as ``x`` type (types are also stripped down from the runtime``. Doing for instance ``x as string`` (when x can be undefined) won't magically convert the object that can be undefined (marked with ``?`` as you know) to a string in real Javascript, TypeScript will just omit that it can be undefined but it won't do any conversion under the hood. That's why we still need to do the boolean conversion, so in runtime javascript the object is really a boolean and not a string as TypeScript started considering since we explicitly told it so.

Reply:
Okay I understand how the `as` works in TS, that's why I was confused. I thought it'd convert the data in it and all. Thanks!

Also on the copy by reference, np, I just thought initially we'd copy but that makes more sense